lib/fluent/plugin_helper/http_server/request.rb

```diff
@@ -35,7 +35,12 @@ def headers
         end
 
         def query
-          @query_string && CGI.parse(@query_string)
+          if @query_string
+            hash = Hash.new { |h, k| h[k] = [] }
+            URI.decode_www_form(@query_string).each_with_object(hash) do |(key, value), h|
+              h[key] << value
+            end
+          end
```

Comment:
Thanks!
Could you please add a comment that this code is to maintain compatibility with `CGI#parse`?

Reply:
added https://github.com/fluent/fluentd/pull/4962/commits/47895b4535561af10445819cfe51a90b6c859a29